l10n_ch_invoice_reports/models/report.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import io
+from io import BytesIO
+
+import PyPDF2
+
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def merge_pdf_in_memory(self, docs):
+        streams = []
+        writer = PyPDF2.PdfFileWriter()
+        for doc in docs:
+            if doc:
+                current_buff = BytesIO()
+                streams.append(current_buff)
+                current_buff.write(doc.read())
+                current_buff.seek(0)
+                reader = PyPDF2.PdfFileReader(current_buff)
+                for page in range(reader.getNumPages()):
+                    writer.addPage(reader.getPage(page))
+            else:
+                writer.addBlankPage()
+        buff = BytesIO()
+        try:
+            # The writer close the reader file here
+            writer.write(buff)
+            return buff.getvalue()
+        except IOError:
+            raise
+        finally:
+            buff.close()
```

Comment:
```suggestion
            buff.close()
            current_buff.close()
```

Reply:
current_buff won't work because it is created in if-block so variable won't be visible here.